pkg/cmd/server/kubernetes/master/controller/config.go

```diff
@@ -0,0 +1,61 @@
+package controller
+
+import (
+	kubecontroller "k8s.io/kubernetes/cmd/kube-controller-manager/app"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+)
+
+// KubeControllerConfig is the runtime (non-serializable) config object used to
+// launch the set of kube (not openshift) controllers.
+type KubeControllerConfig struct {
+	RecyclerImage string
+
+	// TODO the scheduler should move out into its own logical component
+	SchedulerControllerConfig SchedulerControllerConfig
+}
+
+// TODO in 3.7, CloudProvider is on the context
+func (c KubeControllerConfig) GetControllerInitializers(cloudProvider cloudprovider.Interface) (map[string]kubecontroller.InitFunc, error) {
+	ret := kubecontroller.NewControllerInitializers()
+
+	// Remove the "normal" resource quota, because we run it as an openshift controller to cover all types
+	// TODO split openshift separately so we get upstream initialization here
+	delete(ret, "resourcequota")
+	// "serviceaccount-token" is used to create SA tokens for everyone else.  We special case this one.
+	delete(ret, "serviceaccount-token")
+
+	// TODO once the cloudProvider moves, move the configs out of here to where they need to be constructed
+	persistentVolumeController := PersistentVolumeControllerConfig{
+		RecyclerImage: c.RecyclerImage,
+		// TODO: In 3.7 this is renamed to 'Cloud' and is part of kubernetes ControllerContext
```

Comment:
guess we can remove this todos as you have todo in func godoc (which btw. should be in godoc format ;-)